src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorEngine.java

```diff
@@ -0,0 +1,263 @@
+     * Note: The returned list is unmodifiable.
+     */
+    public List<GenotypeAnnotation> getGenotypeAnnotations() {
+        return Collections.unmodifiableList(genotypeAnnotations);
+    }
+
+    /**
+     * Returns the list of info annotations that will be applied.
+     * Note: The returned list is unmodifiable.
+     */
+    public List<InfoFieldAnnotation> getInfoAnnotations() {
+        return Collections.unmodifiableList(infoAnnotations);
+    }
+
+    /**
+     * Returns the set of desciptions to be added to the VCFHeader line (for all annotations in this engine).
+     */
+    public Set<VCFHeaderLine> getVCFAnnotationDescriptions() {
+        final Set<VCFHeaderLine> descriptions = new HashSet<>();
+
+        for ( final InfoFieldAnnotation annotation : infoAnnotations) {
+            descriptions.addAll(annotation.getDescriptions());
+        }
+        for ( final GenotypeAnnotation annotation : genotypeAnnotations) {
+            descriptions.addAll(annotation.getDescriptions());
+        }
+        for ( final String db : variantOverlapAnnotator.getOverlapNames() ) {
+            if ( VCFStandardHeaderLines.getInfoLine(db, false) != null ) {
+                descriptions.add(VCFStandardHeaderLines.getInfoLine(db));
+            } else {
+                descriptions.add(new VCFInfoHeaderLine(db, 0, VCFHeaderLineType.Flag, db + " Membership"));
+            }
+        }
+
+        return descriptions;
+    }
+
+    /**
+     * Annotates the given variant context - adds all annotations that satisfy the predicate.
+     */
+    public VariantContext annotateContext(final FeatureContext features,
+                                          final ReferenceContext ref,
+                                          final VariantContext vc,
+                                          final Map<String,PerReadAlleleLikelihoodMap> perReadAlleleLikelihoodMap,
+                                          final Predicate<VariantAnnotation> addAnnot) {
+        Utils.nonNull(features, "features cannot be null");
+        Utils.nonNull(vc, "vc cannot be null");
+        final Map<String, Object> annotValues = new LinkedHashMap<>(vc.getAttributes());
+
+        for ( final InfoFieldAnnotation annotationType : infoAnnotations) {
+            if (addAnnot.test(annotationType)){
+                final Map<String, Object> annotationsFromCurrentType = annotationType.annotate(ref, vc, perReadAlleleLikelihoodMap);
+                if ( annotationsFromCurrentType != null ) {
+                    annotValues.putAll(annotationsFromCurrentType);
+                }
+            }
+        }
+        final VariantContextBuilder builder = new VariantContextBuilder(vc).attributes(annotValues);
+
+        // annotate genotypes, creating another new VC in the process
+        final VariantContext annotated = builder.genotypes(annotateGenotypes(ref, vc, perReadAlleleLikelihoodMap, addAnnot)).make();
```

Comment:
This is the reverse of the order of operations in GATK3, where genotypes were annotated first and the info field second. Will this change anything?